### ISSUE TITLE
PR: Floors not exporting

### ISSUE THREAD

**Issue by thumDer** [Autodesk/revit-ifc#759]
### Problem Description

Floors not exporting to IFC.
As per the export settings all Floor categories are mapped to `IfcSlab(Type)`, and their corresponding enums. We've tried both with and without "Type", with and without the enums. `IfcSlab` is enabled in the _IFC Entities to Export_ dialog.

We've found the following workarounds, but these are unacceptable in a production environment:
- All Floor categories are mapped to `IfcBuildingElementProxy`
Unfortunately this will render the _Export to IFC As_, and _IFC Predefined Type_ parameters ineffective. Whatever their values are all elements end up `IfcBuildingElementProxy`
- _Export to IFC_ or _Export Type to IFC_ to be set to **Yes** instead of **Default** manually for all types / instances

This is a very inconsistent behaviour. Some of our PCs work just fine, others have this issue. All of them have Revit 24.2.0.63 and IFC Exporter 24.2.0.49

We also had this issue before in Revit 2022, but only on one computer, but now with Revit 2024 it is much more common.

Since there are no logs about the export process we are stuck with troubleshooting.

### Revit Version

2024.0.x

### IFC for Revit Addon Version

24.x.x

### Windows Version

11 22H2

**Comment by AngelVelezSosa** [comment]
Hello, sorry for your issues.  This is not a generic issue (as evidenced by the fact that it works on some of your machines), so we would need access to your files to see if we can reproduce the problem.  It doesn't (and in fact, we prefer if it isn't) have to e your entire model - it can just be one floor that by itself exports on some of your machines and not others.  If you can't narrow it down that way, then sending a file and marking the problematic floor would at least help.  You shouldn't send it here - send a link to my Autodesk address and I'll get it in our internal system.

**Comment by thumDer** [comment]
We tested the affected computers with a minimal empty model, that only has 2 floors modeled, and the issue persist. So I'm almost 100% sure that it isn't model related. It would be really good if we had some debug logs.
We plan to do additional tests on multiple computers, and in multiple Revit versions, because one of the affected computers have the same issuen in both Revit 2022 and Revit 2023 with the most recent IFC Exporters.

We double checked the IFC Files, and it is not only the geometries that are missing, its the complete entity.

**Comment by AngelVelezSosa** [comment]
And everyone is on the same language of Revit?  That's an issue before Revit 2025 (or maybe Revit 2024.X).  I don't really know what environment issues could force the floors not to export.  But have you at least checked the settings on the "bad" computer?

**Comment by thumDer** [comment]
Yes, we all use en-us. The computers supposed to be the same, more or less. Branded Lenovos, with managed installations. I will have more info about the frequency and the specs.

**Comment by AngelVelezSosa** [comment]
I can't imagine the spec would matter for not being to export 2 floors.  Maybe there is some issue getting to the network location?  There has to be something generically wrong with the "bad" machine, whatever it is.

**Comment by thumDer** [comment]
We did some investigation

We've created a test model in versions 2022, 2023, 2024, which had 3 floor instances, one had no per type override, one had IfcSlab.BASESLAB override and the last had IfcSlab.ROOF override
The category mapping had IfcSlab for all Floor categories

from the 3 revit versions we exported 2 ifc files, using ifc2x3 and ifc4 versions, with the default configs

out of 7 computers only one could export the elements, for all the 6 ifc models, 1 exported elements only from revit 2023, the other 5 computers exported empty ifc files

All computers had the same ifc exporter versions

The only workaround that worked, is to set the Export Type to IFC parameter to Yes instead of Default for all the Floor types.

This seems to be a critical bug, that affects most of our users.

**Comment by thumDer** [comment]
I've also found a very similar issue on the forums: https://forums.autodesk.com/t5/revit-architecture-forum/rvt23-ifc-export-problem-slab-site/td-p/12136390
What got me thinking is I assume the user posted that issue is also hungarian from his name, and however all of our Revit installs are en-us, the host OS (Win 11) is hungarian for everyone EXCEPT the only computer that managed to install correct IFCs.
So, this is strange as hell, but it seems that this issue only affects hungarian localized WINDOWS environments?!

**Comment by AngelVelezSosa** [comment]
I still suggest you go through proper Autodesk support for this.  They may have better resources for trying to figure out why it sometimes works.  There clearly has to be _something_ different in the steps otherwise it would never work.

**Comment by thumDer** [comment]
First I will try to change the system language settings on one of the computers to see if it makes any difference. Honestly, I don't really have positive experience with the support, especially in technically complicated issues, but can't hurt.
Can you share some additional insights about the issues regarding the Revit language differences?

**Comment by AngelVelezSosa** [comment]
The only "old" issue was the export mapping table (before Revit 2025) could be in the wrong language, and export nothing.  There could also be issues with encoding, ANSI vs UTF-8, but again those should either all work or all not.  If the computers had different add-ins installed, that could be a thing.  There has to be *something* different.

**Comment by thumDer** [comment]
After testing my hypothesis, I can confirm with 100% certainty that the windows language setting is the cause of this particular problem. After changing the system language, the export succeeds on a computer where it failed before the language change. Nothing was changed except the language.

Here are the before & after exported IFCs:
https://inexkft-my.sharepoint.com/:f:/g/personal/tamas_deri_in-ex_hu/ErjnAYsPr-VDiNsWIHQuby0BAj210MS7GaZIvOp7UPKDrQ?e=eSrRVr

**Comment by AngelVelezSosa** [comment]
I can't access that sharepoint, but is the issue that one IFC file is very tiny (very little in it) and the other one is "complete"?  And you are using a version of Revit before 2025?  If so, then it may just be that you need an export mapping table for each system language, because everything is basically set not to export.  Alternatively, if it is dependent on the language, I wonder if there are categories with characters other than a-z and A-Z that cause the encoding of the mapping file ot have issues.

**Comment by thumDer** [comment]
Sorry, I've just updated the link. The issue is one ifc is empty (has no building elements), while the other has all the floors from the Revit model.
Our Revit is English, so all our mapping files are in english, and have no other characters just the ones you mentioned.

**Comment by batyitam** [comment]
I can confirm for other IFC 4 relate export issue (Ifc export as is onverruled to buildingelementproxy for example roof, slab, stair and some other random elements). Other elements exported normally. If I switch to english language (windows 10) the export works as expected.

**Comment by ArtemHnatko** [comment]
Thank you for the detailed report and for pinpointing the root cause.

We investigated the described behavior and confirmed the following:

- The issue is caused by .NET's `StreamReader` reading `exportlayers-ifc-IAI.txt` (UTF-16 LE) without an explicit encoding, which corrupts category name lookups when the Windows system code page is non-English (e.g., CP1250). This silently drops Floors, Roofs, Slabs, and Stairs from IFC output with no error or warning.
- The behavior was confirmed on Revit 2022–2024 with both IFC2x3 and IFC4 schemas.
- [Autodesk KB](https://www.autodesk.com/support/technical/article/caas/sfdcarticles/sfdcarticles/Revit-IFC-file-not-exported-or-contains-no-objects-after-Revit-interface-language-change.html) confirms the issue was resolved in **Revit 2025**.
- We retested on **Revit 2026** with a Windows system locale of `windows-1250` (CP1250) and an English Revit UI — floors exported correctly. The fix holds in currently supported versions.

**Conclusion:** This issue was addressed in Revit 2025 and does not reproduce in Revit 2026. We are closing this issue.

If you are still experiencing missing elements on Revit 2025 or later, please reopen and include your Revit build number, IFC exporter version, and Windows system locale (`[System.Text.Encoding]::Default.WebName` in PowerShell). In the meantime, the workaround for Revit 2022–2024 is to delete `C:\ProgramData\Autodesk\RVT <year>\exportlayers-ifc-IAI.txt` and regenerate it via **File → Export → Options → IFC Options → Standard**.

Thank you for contributing to the project.
